domain/secret/service_test.go

```diff
@@ -58,38 +60,62 @@ func (s *serviceSuite) SetUpTest(c *gc.C) {
 	c.Assert(err, jc.ErrorIsNil)
 }
 
-func (s *serviceSuite) setup(c *gc.C) *gomock.Controller {
-	ctrl := gomock.NewController(c)
-	s.secretBackendState = secret.NewMockSecretBackendState(ctrl)
+func (s *serviceSuite) TestDeleteSecretInternal(c *gc.C) {
+	defer s.setupMocks(c).Finish()
 
-	s.svc = service.NewSecretService(
-		state.NewState(func() (database.TxnRunner, error) { return s.ModelTxnRunner(c, s.modelUUID.String()), nil }, loggertesting.WrapCheckLog(c)),
-		s.secretBackendState,
-		loggertesting.WrapCheckLog(c),
-		service.SecretServiceParams{},
-	)
+	s.secretBackendState.EXPECT().AddSecretBackendReference(gomock.Any(), nil, s.modelUUID, gomock.Any())
+	uri := s.createSecret(c, map[string]string{"foo": "bar"}, nil)
 
-	return ctrl
+	err := s.svc.DeleteSecret(context.Background(), uri, service.DeleteSecretParams{
```

Comment:
Why these changes on the secret domain?

Reply:
Secrets requires an application and for the life of me couldn't work out where it was, so I refactored the methods and functions to make it simpler to read.